test/trace.spec.ts

```diff
@@ -45,6 +45,59 @@ it('should record trace', test => test.fixme(), async ({browser, testInfo, serve
   expect(gotoEvent.pageId).toBe(pageId);
   expect(gotoEvent.value).toBe(url);
 
+  const resourceEvent = traceEvents.find(event => event.type === 'resource' && event.url.endsWith('/frames/style.css')) as trace.NetworkResourceTraceEvent;
+  expect(resourceEvent).toBeTruthy();
+  expect(resourceEvent.contextId).toBe(contextId);
+  expect(resourceEvent.pageId).toBe(pageId);
+  expect(resourceEvent.method).toBe('GET');
+  expect(resourceEvent.status).toBe(200);
+  expect(resourceEvent.requestHeaders).toBeTruthy();
+  expect(resourceEvent.requestHeaders.length).toBeGreaterThan(0);
+  expect(resourceEvent.requestSha1).toBe('none');
+
   expect(gotoEvent.snapshot).toBeTruthy();
   expect(fs.existsSync(path.join(traceDir, 'resources', gotoEvent.snapshot!.sha1))).toBe(true);
 });
+
+it('should record trace with POST', async ({browser, testInfo, server}) => {
+  const traceDir = testInfo.outputPath('trace');
+  const context = await browser.newContext({ _traceDir: traceDir } as any);
+  const page = await context.newPage();
+  const url = server.PREFIX + '/trace-resources.html';
+  await page.goto(url);
+  await page.click('text=Download');
+  await context.close();
```

Comment:
I think this will solve the race condition:
```js
await page.waitForSelector(`#response-status:text("404")`);
```

Reply:
Hey, I added this fix, but in general, isn't Playwright supposed to wait for network requests to resolve, before finishing actions such as click? This fix will solve the test, but I assume there is a chance that the Network Resource will be assigned to `waitForSelector` action in some cases, right? Isn't that a potential bug somewhere deeper in Playwright?

Reply:
There is a balance to strike here. We do not wait for every possible xhr to finish, otherwise we'll wait forever. Usually, the next action will wait for the logical result of the previous one (for example it will try to click the element that appears as a result of the previous action). However, we do recommend manual waiting in the less common cases.

Reply:
Ok makes sense.